### PR TITLE
Temporary fix of expand module intention around crate roots

### PR DESCRIPTION
**Pull request by pavel-v-chernykh**
Temporary fix of expand module intention around crate roots
Small refactoring

### CODE REVIEW

File: src/org/rust/lang/intentions/ExpandModule.kt [hunk 1]
```diff
@@ -8,15 +8,18 @@ import com.intellij.psi.PsiFile
...
 
 class ExpandModule : IntentionAction {
+    private val DIRECTORY_OWNER_FILE_NAMES = setOf(MOD_RS, MAIN_RS, LIB_RS)
 
     override fun getFamilyName() = "Expand module structure"
     override fun startInWriteAction() = true
     override fun getText() = "Expand Module"
     override fun isAvailable(project: Project, editor: Editor?, file: PsiFile?) =
-        file is RustFileImpl && file.name != MOD_RS
+        file is RustFileImpl && file.name !in DIRECTORY_OWNER_FILE_NAMES
```
Comment (matklad) [inline src/org/rust/lang/intentions/ExpandModule.kt]: It is better use `https://github.com/alexeykudinkin/intellij-rust/pull/150/files#diff-221e21326806663a21f37048add38cd3R27`.

Reasons: DRY and logic for determining if a module is a dir owner **will** change

File: src/org/rust/lang/core/psi/util/RustModExtensions.kt [hunk 2]
```diff
@@ -7,6 +7,8 @@ import org.rust.lang.core.psi.RustModItem
 
 
 val MOD_RS = "mod.rs"
+val MAIN_RS = "main.rs"
+val LIB_RS = "lib.rs"
```
Comment (matklad) [inline src/org/rust/lang/core/psi/util/RustModExtensions.kt]: I don't think it is necessary to extract this constants, they are a private implementation detail, unlike `MOD_RS`

**Comment by pavel-v-chernykh** [comment]
@matklad I'm confused, I would rather close the PR. It just spends your time. Thanks for comments.

**Comment by matklad** [comment]
I do agree that some refactoring is necessary here to make expand module and nonexisting module actions more natural. For this, I think it is necessary

*  make `isDirectoryOwner` public and provide some way to get a mod from a PSI file. This way, you can simplify check in expand to `if (file.rustMod != null && !file.rustMod.isDirectoryOwner)`. The code for extracting mod from file is [here](https://github.com/pavel-v-chernykh/intellij-rust/blob/29f08790aed997d2fc9dc81270909909ee6662da/src/org/rust/lang/core/psi/util/RustModExtensions.kt#L35). You need to extract this as an extension method for a PSIFile.

* change moduleFile in such a way that in case of `NotFound` it provides a hint for the file, where the module should be. This will help with nonexisting module because you won't have to reconstruct the file path yourself (it is not trivial for nested modules). So the change is `class NotFound(suggestedPath: String?)`.

I also like the simplification of `return when (variants.size) ` part :)

**Comment by pavel-v-chernykh** [comment]
> I also like the simplification of return when (variants.size) part :)

@matklad feel free to use it :smirk:

**Pull request closed without merging**